### DISCUSSION TITLE
Anyons class

### DISCUSSION THREAD

**Discussion by aburousan** [quantum-visualizations/qmsolve#8] [Ideas]
Hi, I am a student doing physics hons. I was recently writing a project and discovered about "Anyons". I was curious if it is possible to implement it in this solver?
You can find the information of anyons here - https://drive.google.com/file/d/1cgqWuUhzab-suRP2NC967XC5siBkOd-p/view?usp=sharing
It's mechanics is simply psi(r1,r2) = exp(i*pi*v) psi(r2,r1) , where v = 0 is for boson and v = 1 is for fermion.

**Comment by rafael-fuente** [comment]
Nice suggestion. It should be perfectly possible.  You just need to replace line 28 of [two_fermions.py ](https://github.com/quantum-visualizations/qmsolve/blob/main/qmsolve/particle_system/two_fermions.py)

```
eigenstate_tmp = (eigenvectors[i] - eigenvectors[i].swapaxes(0,1))/np.sqrt(2)
```

by

```
v= 0.5 # v = 0 for a boson and v = 1 for a fermion.
eigenstate_tmp = (eigenvectors[i] + np.exp(1j*np.pi*v) *  eigenvectors[i].swapaxes(0,1))/np.sqrt(2)
```
And the fermion solver now would be an anyon solver with v = 0.5.

**Reply by aburousan** [reply]
I have done that but i am getting the energy levels are for interacting anyons and fermions. They should be different. As shown in the article linked before(page- 17)
![Screenshot (208)](https://user-images.githubusercontent.com/60939745/122000168-0670a480-cdcc-11eb-8d8d-edc1d317cdf2.png)
In this image, ξ = 0 for boson and  ξ = π for fermion. As you can see
![Screenshot (209)](https://user-images.githubusercontent.com/60939745/122000599-b8a86c00-cdcc-11eb-9cff-768dce256144.png)
the energy levels should atleast be different.

**Reply by rafael-fuente** [reply]
Where are these pages in the lecture you linked? I didn't found them on page 17.

Sorry for the quick answer, I never studied Anyons and what I said isn't enough because the Hamiltonian also changes.
I guess that implementing an Anyons class, it's going to require a substantial amount of work, but it should still be perfectly possible.
Because the Hamiltonian depends on ξ,  we need to replace the way it's built. 
If the Hamiltonian can be expressed as in the form H(ξ) + V, then you need to write the correct form of H(ξ) at get_kinetic_matrix method, which currently for the two-particle class, it returns the matrix of the discretized kinetic operator expressed on a cartesian basis (Tx1 ⊗ I  +  I ⊗ Tx2). To achieve it, you'll need to adequately discretize H(ξ) using finite differences.

There are still more basic aspects of the solver to work on (such as implementing spin), so I would not be willing to work on this feature yet. However, if you want to contribute by implementing it, you are completely welcome.

**Reply by aburousan** [reply]
Just sharing in case you want to read:
Anyone Hamilton: https://arxiv.org/abs/1807.10904

**Comment by aburousan** [comment]
Thank you for your reply. I will try to do that if I am able to do that
then I will let you know.
